File: 2024/DS/Polimorfismo (Conta)/Conta.java

public class Conta {
    protected String nome;
    protected String CPF;
    protected String numeroConta;
    protected double saldo;
 
    public Conta(String nome, String CPF, String numeroConta, double saldo) {
        this.nome = nome;
        this.CPF = CPF;
        this.numeroConta = numeroConta;
        this.saldo = saldo;
    }
 
    public double depositar(double quantia) {
        saldo += quantia;
        return saldo;
    }
 
    public double sacar(double quantia) {
        if (quantia <= saldo) {
            saldo -= quantia;
        } else {
            System.out.println("Saldo insuficiente!");
        }
        return saldo;
    }
 
    public void imprimirSaldo() {
        System.out.println("Saldo atual: " + saldo);
    }
 
    public double getSaldo() {
        return saldo;
    }
}

File: 2025/Qualidade e teste de software/Teste de classe conta/model/Conta.js

class Conta {
    constructor(agencia, numero, senha, saldo = 0) {
      this.agencia = agencia;
      this.numero = numero;
      this.senha = senha;
      this.saldo = saldo;
    }
  
    autenticar(agencia, numero, senha) {
      return (
        this.agencia === agencia &&
        this.numero === numero &&
        this.senha === senha
      );
    }
  
    consultarSaldo(agencia, numero, senha) {
      if (!this.autenticar(agencia, numero, senha)) {
        throw new Error('Autenticação inválida');
      }
      return this.saldo;
    }
  
    sacar(agencia, numero, senha, valor) {
      if (!this.autenticar(agencia, numero, senha)) {
        throw new Error('Autenticação inválida');
      }
      if (valor > this.saldo) {
        throw new Error('Saldo insuficiente');
      }
      this.saldo -= valor;
    }
  
    depositar(agencia, numero, senha, valor) {
      if (!this.autenticar(agencia, numero, senha)) {
        throw new Error('Autenticação inválida');
      }
      this.saldo += valor;
    }
  }
  
  module.exports = Conta;
  